internal/server/v1/firehose/mappings.go

```diff
@@ -241,8 +241,10 @@ func (fc *firehoseConfigs) toConfigStruct(prj *shieldv1beta1.Project) (*structpb
 	const defaultState = "RUNNING"
 
 	metadata := prj.GetMetadata().AsMap()
-	telegrafConf, ok := metadata["telegraf"].(map[string]interface{})
-	if !ok || len(telegrafConf) == 0 {
+	var telegrafConf map[string]interface{}
+	telegrafConfString, ok := metadata["telegraf"].(string)
+	err := json.Unmarshal([]byte(telegrafConfString), &telegrafConf)
+	if !ok || len(telegrafConf) == 0 || err != nil {
```

Comment:
unmarshal should be inside of `if ok {` check i think.. that way, we can at-least log the error in warning.

```golang
var telegrafConf map[string]interface{}
telegrafConfString, ok := metadata["telegraf"].(string)
if ok {
   if err := json.Unmarshal([]byte(telegrafConfString), &telegrafConf); err != nil {
       log.Warnf("telegraf config is invalid")
   } 
}

if len(telegrafConf) == 0 {
    telegrafConf = map[string]interface{}{"enabled": false}
}
```

Reply:
makes sense

Reply:
Sure that sounds good. As for the logging part for now shall we can ignore the Unmarshal error since we are not logging anywhere apart from handlers and also not taking any action on this error?